scaleway/table_scaleway_invoices.go

```diff
@@ -0,0 +1,259 @@
+	}
+
+	amount := float64(money.Units) + float64(money.Nanos)/1e9
+	if d.ColumnName == "total_discount_amount" {
+		return math.Abs(amount), nil
+	}
+	return amount, nil
+}
+
+func extractCurrency(ctx context.Context, d *transform.TransformData) (interface{}, error) {
+	plugin.Logger(ctx).Debug("extractCurrency", "field", d.ColumnName, "raw_value", d.Value)
+
+	if d.Value == nil {
+		plugin.Logger(ctx).Warn("extractCurrency: nil value", "field", d.ColumnName)
+		return nil, nil
+	}
+
+	switch v := d.Value.(type) {
+	case *scw.Money:
+		if v == nil {
+			return nil, nil
+		}
+		return v.CurrencyCode, nil
+	default:
+		plugin.Logger(ctx).Warn("extractCurrency: unexpected type", "type", fmt.Sprintf("%T", d.Value))
+		return nil, nil
+	}
+}
+
+type invoicesItem struct {
+	ID               string              `json:"id"`
+	OrganizationID   string              `json:"organization_id"`
+	OrganizationName string              `json:"organization_name"`
+	StartDate        *time.Time          `json:"start_date"`
+	StopDate         *time.Time          `json:"stop_date"`
+	IssuedDate       *time.Time          `json:"issued_date"`
+	DueDate          *time.Time          `json:"due_date"`
+	TotalUntaxed     *scw.Money          `json:"total_untaxed"`
+	TotalTaxed       *scw.Money          `json:"total_taxed"`
+	TotalDiscount    *scw.Money          `json:"total_discount"`
+	TotalUndiscount  *scw.Money          `json:"total_undiscount"`
+	TotalTax         *scw.Money          `json:"total_tax"`
+	Type             billing.InvoiceType `json:"type"`
+	Number           int32               `json:"number"`
+	State            string              `json:"state"`
+	BillingPeriod    *time.Time          `json:"billing_period"`
+	SellerName       string              `json:"seller_name"`
+}
+
+func listScalewayInvoices(ctx context.Context, d *plugin.QueryData, _ *plugin.HydrateData) (interface{}, error) {
+	// Get client configuration
+	client, err := getSessionConfig(ctx, d)
+	if err != nil {
+		plugin.Logger(ctx).Error("scaleway_invoices.listScalewayInvoices", "connection_error", err)
+		return nil, err
+	}
+
+	// Check if the client is properly configured
+	if client == nil {
+		return nil, fmt.Errorf("scaleway client is not properly configured")
+	}
+
+	billingAPI := billing.NewAPI(client)
+
+	// Prepare the request
+	req := &billing.ListInvoicesRequest{}
+
+	// Get the organization_id from the config
+	scalewayConfig := GetConfig(d.Connection)
+	var organizationID string
+	if scalewayConfig.OrganizationID != nil {
+		organizationID = *scalewayConfig.OrganizationID
+	}
+
+	// Check if organization_id is specified in the query
+	if d.EqualsQualString("organization_id") != "" {
+		organizationID = d.EqualsQualString("organization_id")
+	}
+
+	// Log a warning if organization_id is not provided by either config or query
+	if organizationID == "" {
+		plugin.Logger(ctx).Warn("scaleway_invoices.listScalewayInvoices", "warning", "No organization_id provided in config or query")
```

Comment:
It might be more appropriate to return an error instead of just a warning message in this scenario.

Could you please clarify how the API behaves when the Organization ID is not provided? 
Does it default to returning results for the current Organization associated with the provided access key and secret key?

If the API does not return an error in such cases, I kindly suggest removing this block of code for simplicity and consistency with the API behavior.